include/zephyr/llext/symbol.h

```diff
@@ -113,32 +125,45 @@ struct llext_symtable {
  *
  * @param x Extension symbol to export to the base image
  */
-#define LL_EXTENSION_SYMBOL(x) Z_LL_EXTENSION_SYMBOL(x)
+#define LL_EXTENSION_SYMBOL(x) LL_EXTENSION_SYMBOL_NAMED(x, STRINGIFY(x))
 
 /** @cond ignore */
 #if defined(LL_EXTENSION_BUILD)
-/* Extension build: EXPORT_SYMBOL maps to LL_EXTENSION_SYMBOL */
-#define Z_EXPORT_SYMBOL(x) Z_LL_EXTENSION_SYMBOL(x)
+/* Extension build: EXPORT_SYMBOL_NAMED maps to LL_EXTENSION_SYMBOL_NAMED */
+#define Z_EXPORT_SYMBOL_NAMED(sym_ident, sym_name)                                      \
+	Z_LL_EXTENSION_SYMBOL_NAMED(sym_ident, sym_name)
 #elif defined(CONFIG_LLEXT_EXPORT_BUILTINS_BY_SLID)
-/* SLID-enabled LLEXT application: export symbols, names in separate section */
-#define Z_EXPORT_SYMBOL(x)							\
-	static const char Z_GENERIC_SECTION("llext_exports_strtab") __used	\
-		x ## _sym_name[] = STRINGIFY(x);				\
-	static const STRUCT_SECTION_ITERABLE(llext_const_symbol, x ## _sym) = {	\
-		.name = x ## _sym_name, .addr = (const void *)&x,		\
+/* LLEXT- and SLID-enabled application: export symbols with names in separate section */
+#define Z_EXPORT_SYMBOL_NAMED(sym_ident, sym_name)					\
+	static const char Z_GENERIC_SECTION("llext_exports_strtab") __used	        \
+		sym_ident ## _sym_name[] = sym_name;				        \
+	static const STRUCT_SECTION_ITERABLE(llext_const_symbol, sym_ident ## _sym) = {	\
+		.name = sym_ident ## _sym_name, .addr = (const void *)&sym_ident,	\
 	}
 #elif defined(CONFIG_LLEXT)
-/* LLEXT application: export symbols */
-#define Z_EXPORT_SYMBOL(x)							\
-	static const STRUCT_SECTION_ITERABLE(llext_const_symbol, x ## _sym) = {	\
-		.name = STRINGIFY(x), .addr = (const void *)&x,			\
+/* LLEXT-enabled application: export symbols */
+#define Z_EXPORT_SYMBOL_NAMED(sym_ident, sym_name)					\
+	static const STRUCT_SECTION_ITERABLE(llext_const_symbol, sym_ident ## _sym) = {	\
```

Comment:
I see why it's been done this way (`sym_name` is a string), but wouldn't using `sym_ident` to form the `llext_const_symbol`'s name prevent exporting the same symbol twice under different names? (at least if both are in the same translation unit)

I think taking the `sym_name` in token form and `STRINGIFY()`ing it in the macro would be acceptable, if that can help solve this issue.

Same remark for SLID version of the macro.

Reply:
Why would you have to export the same symbol twice with two names? One should be enough for everybody! :hand_over_mouth: 

... seriously, this PR wanted to transparently rename symbols when appropriate, so I wasn't thinking about the use case you mentioned (there's always one `EXPORT_SYMBOL` for each identifier). 
It makes sense though, I will have to think about using the `sym_name` - I remember there was a problem that made me use strings and not tokens, but I'm not sure what that was.